File: Two-pointer/1806번 - 부분합/1806.cpp
#include <iostream>

using namespace std;

int n, s;
int seq[100001] = { 0, };

void input() {
    cin >> n >> s;
    for(int i = 0; i < n; i++)
        cin >> seq[i];
}

void solve() {
    int st = 0, end = 0;
    int sum = seq[0];
    int ans = 1e9;
    
    while(st <= end && end < n) {
        if(sum < s) {
            end++;
            sum += seq[end];
        }
        else {
            ans = min(ans, end - st + 1);
            sum -= seq[st];
            st++;
        }
    }
    
    cout << (ans != 1e9 ? ans : 0) << '\n';
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    input();
    solve();
    
    return 0;
}

File: Two-pointer/1806번 - 부분합/1806.py
import sys

input = sys.stdin.readline

n, s = map(int, input().split())
seq = list(map(int, input().split()))

def solve():
    st = end = 0
    seq_sum = seq[0]
    ans = 1e9
    
    while st <= end <= n - 1:
        if seq_sum < s:
            end += 1
            if end == n: 
                break
            seq_sum += seq[end]
        else:
            ans = min(ans, end - st + 1)
            seq_sum -= seq[st]
            st += 1
    
    print(ans if ans != 1e9 else 0)

if __name__ == '__main__':
    solve()